### PR TITLE
Split datetime into three UI types

### LINKED ISSUE #960: Split Date & Time data type into three Mathesar types
## Problem

Based on the new definition of [Mathesar Data Types](https://wiki.mathesar.org/en/engineering/architecture/mathesar-types), it makes sense to split up the Date & Time type into three types:
- Date
- Time
- Date & Time

## Proposed solution

We need to update the data type design specs to create specs for `Date`  and  `Time` and update the `Date & Time` specs to remove the options for date-only and time-only database options.

## Additional context

- The [Data Types "Concepts" wiki page](https://wiki.mathesar.org/product/concepts/data-types), aimed at a general audience.
- The [Mathesar Types architecture wiki page](https://wiki.mathesar.org/en/engineering/architecture/mathesar-types), aimed at Mathesar core team members.
- https://github.com/centerofci/mathesar/discussions/959#discussioncomment-1927668

### PR DESCRIPTION
**Pull request by dmos62**
Fixes #960

Changes the definition of Mathesar/UI types. What was before `datetime` UI type, now is `date`, `time` and `datetime` types.

**Technical details**

Below description is outdated. See this comment: https://github.com/centerofci/mathesar/pull/1093#issuecomment-1055575527

@pavish @seancolsen these types' display options are different too. `datetime` initially accepted `format` display option. Now, `datetime` accepts `date_format` and `time_format` display options: `date` accepts `date_format`: and, `time` accepts `time_format`.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dmos62** [comment]
Based on https://github.com/centerofci/mathesar/pull/1089.

**Comment by dmos62** [comment]
@silentninja I made some slight refactors to the display options serializer.

See https://github.com/centerofci/mathesar/pull/1093/files#diff-0cba368aff485c57e63a82055dfe283abac0a640d1896e087ae59d160210270cL105

and https://github.com/centerofci/mathesar/pull/1093/files#diff-0cba368aff485c57e63a82055dfe283abac0a640d1896e087ae59d160210270cR250

Could you give it a quick glance for a sanity check?

**Comment by kgodey** [comment]
@dmos62 can you mark as a draft since this isn't ready to be merged (due to other PRs needing to be merged first?)

**Comment by dmos62** [comment]
@kgodey undrafting is part of the process to mark this as ready for review. Should I undraft and redraft in this case?

**Comment by dmos62** [comment]
Major changes after sync call. What frontend needs to know (first post is now outdated):

- `time`, `date` and `datetime` formats accept only the `format` display option;
- `format` display option is validated to contain no more than 255 characters.

**Comment by kgodey** [comment]
> undrafting is part of the process to mark this as ready for review. Should I undraft and redraft in this case?
> 
> Edit: code review guidelines for authors don't account for the scenario where a PR shouldn't be merged, but can be reviewed.

@dmos62 I would either:
- leave it as draft and set `status` to review, assign reviewers manually, and post in the Code Review channel
- not mark it as ready for review until it's actually ready to be merged and instead remind people to prioritize reviewing the PRs blocking the merge of the current PR.

Please update the code review guidelines in the wiki.

Also, I recommend basing as many PRs as you can off of `master` if what you're working on doesn't depend on functionality from previous PRs. You might already be doing that.

**Comment by dmos62** [comment]
@kgodey I'll update the wiki.

> Also, I recommend basing as many PRs as you can off of master if what you're working on doesn't depend on functionality from previous PRs. You might already be doing that.

I'm already doing that. Most PRs end up implementing something that will affect the other PRs. I also sometimes base off another dev branch to avoid having to resolve merge conflicts.

**Comment by dmos62** [comment]
> @silentninja I made some slight refactors to the display options serializer.
> 
> See https://github.com/centerofci/mathesar/pull/1093/files#diff-0cba368aff485c57e63a82055dfe283abac0a640d1896e087ae59d160210270cL105
> 
> and https://github.com/centerofci/mathesar/pull/1093/files#diff-0cba368aff485c57e63a82055dfe283abac0a640d1896e087ae59d160210270cR250
> 
> Could you give it a quick glance for a sanity check?

@silentninja after implementing changes requested in sync call, the refactor I mentioned has been removed, because most of the code it was refactoring isn't necessary anymore. You might want to take a look anyway.

File: mathesar/api/serializers/shared_serializers.py
Comment (silentninja) [inline mathesar/api/serializers/shared_serializers.py:127]: It would be better to use the `max_length` parameter instead of custom validator, this would help up to throw errors with proper error codes.

Comment (dmos62) [reply mathesar/api/serializers/shared_serializers.py]: Fixed https://github.com/centerofci/mathesar/pull/1093/commits/27f75caa9a17270e0dbfed119001286ec401c181

File: mathesar/api/serializers/shared_serializers.py
Comment (silentninja) [inline mathesar/api/serializers/shared_serializers.py:131]: Based on our [conclusion](https://wiki.mathesar.org/en/meeting-notes/2022-03#h-2022-03-01), other than a character length check, we don't need any additional validators. So the `validators` field  and the validation function should be removed

**Review by silentninja — APPROVED** [review]
Also, remove the `arrow` library` as it is unnecessary. Please make the requested changes and merge.

**Comment by dmos62** [comment]
Removed arrow library and custom time/duration format validation.

**Pull request merged**
